snap/hooks/configure

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+set -eux
+
+
+function snap_logging () {
+    mkdir -p "${SNAP_LOG_DIR}"
+
+    LOG_FILE_PATH="${SNAP_LOG_DIR}/${1}.log"
+
+    function log() {
+        while read -r
+        do
+            echo "$(date) $REPLY" >> "${LOG_FILE_PATH}"
+        done
+    }
+
+    exec 3>&1 1>> >(log) 2>&1
+}
+
+# ----------------------------
+
+function systemd_override () {
+    DAEMON_SYSTEMD_PATH="/etc/systemd/system/snap.${PROJECT_NAME}.daemon.service.d"
+    mkdir -p "${DAEMON_SYSTEMD_PATH}"
+
+    # We are copying the service conf file patch as editing the service file generated by snapd won't survive upgrades
+    cp "${SNAP}"/ops/sys/etc/systemd/override.conf "${DAEMON_SYSTEMD_PATH}"/
+}
+
+function system_conf_override () {
+    # 1. Set the number of open file handles
+    # ulimit -n 1024 -- default in local machine
+    ulimit -n 65535
+
+    # 2. Allow the opensearch user to Disable all swap files:
+    # swapon -a -- default in local machine
+    swapoff -a || true
+
+    # 3. Ensuring sufficient virtual memory: https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html
+    # sysctl -w vm.max_map_count=65530 -- default in local machine
+    sysctl -w vm.max_map_count=262144
+
+    # 4. number of threads opensearch can create:
+    # should be configured automatically if opensearch ran as a service
+
+    # 5. Reduce TCP retransmission timeout = ~6 seconds
+    # sysctl -w net.ipv4.tcp_retries2=15 -- default in local machine
+    sysctl -w net.ipv4.tcp_retries2=5
+}
+
+# -------------------------------
+
+function set_access_restrictions () {
+    if [[ $# -eq 2 ]]; then
+        chmod -R "$2" "$1"
+    fi
+    chown -R snap_daemon:snap_daemon "$1"
+}
+
+function add_folder () {
+    mkdir -p "$1"
+    set_access_restrictions "$1" "$2"
+}
+
+function add_file () {
+    touch "$1"
+    set_access_restrictions "$1" "$2"
+}
+
+function dir_copy_if_not_exists () {
+    cp -R -n -r -p "${SNAP}/$1" "$2"
+
+    if [[ $# -eq 3 ]]; then
+        set_access_restrictions "$2/$1" "$3"
+    else
+        set_access_restrictions "$2/$1"
+    fi
+}
+
+function file_copy () {
+    cp -n -p "${SNAP}/$1" "$2"
+
+    if [[ $# -eq 3 ]]; then
+        set_access_restrictions "$2/$1" "$3"
+    else
+        set_access_restrictions "$2/$1"
+    fi
+}
+
+function set_yaml_property () {
+    if grep -q "^#\?\s*$2" "$1";
+    then
+        sed -i "s@.*$2.*@$3@" "$1"
+    else
+        echo "$3" >> "$1"
+    fi
+}
```

Comment:
I don't know exactly how `stage-snaps` works, but it might be worth using it to include `yq` so that you can do these things a little more precisely with `yq`?